scalameta/inputs/shared/src/main/scala/scala/meta/inputs/Input.scala

```diff
@@ -111,6 +118,19 @@ object Input {
     override def toString = s"Input.Ammonite($input)"
   }
 
+  final case class WithTokenizerOptions private (input: Input)(options: TokenizerOptions)
+      extends Proxy {
+    override def toString = s"Input.WithTokenizerOptions($input, $tokenizerOptions)"
+    override private[meta] def tokenizerOptions = Some(options)
+    override def withoutTokenizerOptions: Input = input.withoutTokenizerOptions
+    override def withTokenizerOptions(implicit options: Option[TokenizerOptions]): Input = options
+      .fold(withoutTokenizerOptions)(x => if (x eq this.options) this else copy()(x))
+  }
+  object WithTokenizerOptions {
+    def apply(input: Input)(implicit options: Option[TokenizerOptions]): Input = options
```

Comment:
I found it a bit cumbersome that you need to wrap options in `Some` here (as well as in the `withTokenizerOptions ` above). I assume it's because the default implicit expose in `TokenizerOptions` has that signature? Could we drop the `Option` wrapper here and there (putting the actual `default` instance as default implicit)?

Reply:
will add an overload

Reply:
#3912